pkg/secrets/secrets.go

```diff
@@ -31,132 +30,142 @@ import (
 	"github.com/hashicorp/vault/api"
 )
 
-const defaultKeyName = "value"
-const secretLocationPrefix = "DAYTONA_SECRET_DESTINATION_"
+const (
+	defaultKeyName           = "value"
+	secretDestinationPrefix  = "DAYTONA_SECRET_DESTINATION_"
+	secretStoragePathPrefix  = "VAULT_SECRET_"
+	secretsStoragePathPrefix = "VAULT_SECRETS_"
+)
 
-// SecretFetcher is responsible for fetching sercrets..
-func SecretFetcher(client *api.Client, config cfg.Config) {
-	locations := prefixSecretLocationDefined()
-	if config.SecretPayloadPath == "" && !config.SecretEnv && len(locations) == 0 {
-		log.Println("No secret output method was configured, will not attempt to retrieve secrets")
-		return
-	}
+// SecretDefinition is used for representing
+// a secret definition input
+type SecretDefinition struct {
+	envkey            string
+	secretID          string
+	secretApex        string
+	outputDestination string
+	paths             []string
+	secrets           map[string]string
+	plural            bool
+}
 
+// SecretFetcher inspects the environment for variables that
+// define secret definitions. The variables are used to guide
+// the SecretFetcher in acquring and outputting the specified secrets
+func SecretFetcher(client *api.Client, config cfg.Config) {
 	log.Println("Starting secret fetch")
-	secrets := make(map[string]string)
+
+	defs := make([]*SecretDefinition, 0)
 
 	ctx := context.Background()
 	parallelReader := NewParallelReader(ctx, client.Logical(), config.Workers)
 
 	envs := os.Environ()
 	// Find where all our secret keys are in vault
-	for _, v := range envs {
-		pair := strings.Split(v, "=")
+	for _, env := range envs {
+		// VAULT_SECRET_WHATEVER=secret/application/thing
+		// VAULT_SECRETS_WHATEVER=secret/application/things
+		// envKey=secretPath
+		pair := strings.Split(env, "=")
 		envKey := pair[0]
-		secretPath := os.Getenv(envKey)
-		if secretPath == "" {
+		apex := os.Getenv(envKey)
+		if apex == "" {
 			continue
 		}
 
-		// Single secret
-		if strings.HasPrefix(envKey, "VAULT_SECRET_") {
-			parallelReader.AsyncRequestKeyPath(secretPath)
-			secretResult := parallelReader.ReadSecretResult()
+		def := &SecretDefinition{
+			envkey:     envKey,
+			secretApex: apex,
+			secrets:    make(map[string]string),
+		}
 
-			err := addSecrets(client, secretResult, secrets)
-			if err != nil {
-				log.Fatalln(err)
+		switch {
+		case strings.HasPrefix(envKey, secretStoragePathPrefix):
+			def.secretID = strings.TrimPrefix(envKey, secretStoragePathPrefix)
+			def.paths = append(def.paths, apex)
+		case strings.HasPrefix(envKey, secretsStoragePathPrefix):
+			def.secretID = strings.TrimPrefix(envKey, secretsStoragePathPrefix)
+			def.plural = true
+		default:
+			continue
+		}
+
+		// look for a corresponding secretDestinationPrefix key.
```

Comment:
If we're going to say that case doesn't matter, I suggest making it not matter at all within the string. For example, this code currently only equates "SUFFIX" and "suffix".

If the case differs in a substring, e.g. "SUFFIX" and "SUFfix", this code wont match. I personally prefer vars to be truly case sensitive. But as it is, it's neither truly case sensitive nor insensitive.

Reply:
Looks like lines changed since my comment. I'll let you figure it out and whether you want to change it :)